iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2432,15 +2477,17 @@ static int InitializeConnection(PMQTTTRANSPORT_HANDLE_DATA transport_data)
     return result;
 }
 
-static STRING_HANDLE buildConfigForUsername(const IOTHUB_CLIENT_CONFIG* upperConfig, const char* moduleId)
+// At handle creation time, we don't have all the fields required for building up the user name (e.g. productID)
+// Build what we can (instead of making separate copies the passed fields) for now.
```

Comment:
```suggestion
// Build what we can (instead of making separate copies of the passed fields) for now.
```

Reply:
haha trust me if my dad saw half the stuff I write he'd be shaking his head for sure